rust/rust-toolchain.toml

```diff
@@ -0,0 +1,2 @@
+[toolchain]
+channel = "nightly"
```

Comment:
I don't want to bump this to requiring the nightly toolchain for such a tiny library. Please remove nightly features from your PR.

Reply:
I see now that there's a note about this on the commit introducing it. I think it would have been preferable to have an inline comment, so it's not lost information when it gets squashed for merge.

It is impossible to search for the metacharacter salad that is that feature flag, and I'm not versed enough in Rust to know where that would be documented; what is that flag doing, and can it be removed?

Or am I completely off-base here and it doesn't matter what toolchain is used.

Reply:
The PR doesn't actually contain newly-used nightly features, `nightly` was just chosen based on my use of the `pattern` feature.

I don't think it changes the version requirements either way(?), given that it compiles in the 2021 edition, and I haven't been using `+nightly` in my usage.

Reply:
I looked more into this; sure seems like a gigantic mess. Should we be using "unstable" features in a library *at all*?

Reply:
The only 'unstable' thing that is being used is the `Pattern` type, which was the simplest way I could figure out how to implement a proper 'search and split' function.
It isn't some form of unstable wizardry (I'm not good enough at Rust for that).

(I actually tried to write the function without it, but I couldn't find a suitable type to take in a Pattern-like function)